File: fjtcmd-hub-generator-cli/docs/generator/demo/product/output/vue/types/api/demo/product.ts
import type { PageDomain, BaseEntity } from "../common";

/** 产品信息配置分页查询参数 */
export interface ProductQueryParams extends PageDomain {
  /** 产品名称 */
  productName?: string;
  /** 显示顺序 */
  orderNum?: number;
  /** 产品状态（0正常 1停用） */
  status?: string;
}

/** 产品信息配置信息 */
export interface Product extends BaseEntity {
  /** 产品id */
  productId?: number;
  /** 父产品id */
  parentId?: number;
  /** 祖级列表 */
  ancestors?: string;
  /** 产品名称 */
  productName?: string;
  /** 显示顺序 */
  orderNum?: number;
  /** 产品状态（0正常 1停用） */
  status?: string;
}

File: fjtcmd-hub-generator-cli/docs/generator/demo/product/output/main/java/com/fjtcmd/hub/demo/domain/Product.java
package com.fjtcmd.hub.demo.domain;

import org.apache.commons.lang3.builder.ToStringBuilder;
import org.apache.commons.lang3.builder.ToStringStyle;
import com.fjtcmd.hub.common.annotation.Excel;
import com.fjtcmd.hub.common.core.domain.TreeEntity;

/**
 * 产品信息对象 sys_product
 * 
 * @author lihd
 * @date 2026-06-11
 */
public class Product extends TreeEntity
{
    private static final long serialVersionUID = 1L;

    /** 产品id */
    private Long productId;

    /** 产品名称 */
    @Excel(name = "产品名称")
    private String productName;

    /** 产品状态（0正常 1停用） */
    @Excel(name = "产品状态", readConverterExp = "0=正常,1=停用")
    private String status;

    public void setProductId(Long productId) 
    {
        this.productId = productId;
    }

    public Long getProductId() 
    {
        return productId;
    }

    public void setProductName(String productName) 
    {
        this.productName = productName;
    }

    public String getProductName() 
    {
        return productName;
    }

    public void setStatus(String status) 
    {
        this.status = status;
    }

    public String getStatus() 
    {
        return status;
    }

    @Override
    public String toString() {
        return new ToStringBuilder(this,ToStringStyle.MULTI_LINE_STYLE)
            .append("productId", getProductId())
            .append("parentId", getParentId())
            .append("ancestors", getAncestors())
            .append("productName", getProductName())
            .append("orderNum", getOrderNum())
            .append("status", getStatus())
            .toString();
    }
}
